DEVELOPERS.md

```diff
@@ -0,0 +1,134 @@
+
+## Classes
+**Locomotive**
+
+* This is where it all begins.
+* A new `Replication` object is created using the `ReplicationFactory` and *replicate* is called on it.
+
+**ReplicationFactory**
+
+* Returns a `Replication` object. The type depends on whether the source table is partitioned or not, and the replication mode specified in the configuration file.
+
+**Replication**
+
+* Either partitioned or unpartitioned.
+* There are 4 replication modes:
+   * `FULL` ← default
+   * `FULL_OVERWRITE`
+   * `METADATA_MIRROR`
+   * `METADATA_UPDATE`
+* Uses a copier based on where the data is coming from and going to:
+   * HDFS or S3 → HDFS, uses `DistCpCopier`
+   * HDFS → S3, uses `S3MapreduceCpCopier`
+   * S3 → S3,  uses `S3S3Copier`
+      * Note: If you are replicating S3 → S3 cross account, *and* you want to assume a role in the target account (see `copier-options.assume-role` in `README.md`), then you must use `S3MapreduceCpCopier`.
+* The data is copied over first (if the mode is `FULL` or `FULL_OVERWRITE`).
+* Then the metadata of the table is updated.
+
+## Types of replication
+There are four types of replication which Circus Train can handle:
+
+* `FULL` ← default
+* `FULL_OVERWRITE`
+* `METADATA_MIRROR`
+* `METADATA_UPDATE`
+
+
+### Full Replication
+**Partitioned**
+
+If the source table has partitions then these and the corresponding data will be copied over to the replica table. After this, the metadata of the table will be updated. 
+
+Otherwise, if the source table has no partitions only the metadata of the table will be updated. 
+
+**Unpartitioned** 
+
+All data from the source is copied over to the replica table, then the metadata is updated.
+
+
+### Full Overwrite Replication
+This replication mode behaves in the same way as `FULL`; however, the corresponding existing replica table, if any, and its underlying data will first be deleted before being replaced with the source table and data. 
+
+This mode is useful in the early stages of the development lifecycle when incompatible schema changes are being made constantly. 
+
+A `DataManipulator` is used to handle the deletion of data. Determining which manipulator to use is handled in the same manner as the [Copier](#copiers), in that there is a `DataManipulatorFactoryManager` which will generate a suitable `DataManipulatorFactory` that returns a `DataManipulator` object. 
+
+### Metadata Mirror Replication 
+Only metadata will be copied (mirrored) from the source to the replica. Replica metadata will not be modified so your source and replica will have the same data location.
+
+*NOTE:* The replica table will be marked as `EXTERNAL`. This is done to prevent accidental data loss when dropping the replica. 
+
+For example, this can be used for copying someone else's metadata into your Hive Metastore without copying the data or to replicate a view. You still need to have access to the data in order to query it.
```

Comment:
```suggestion
Example use case:  this can be used for copying someone else's metadata into your Hive Metastore without copying the data or to replicate a view. You still need to have access to the data in order to query it.
```
(to be consistent with what you've done below)